lib/rex/proto/kerberos/keytab/keytab.rb

```diff
@@ -0,0 +1,151 @@
+# -*- coding: binary -*-
+
+require 'bindata'
+
+module Rex::Proto::Kerberos::Keytab
+  class KeytabCountedOctetString < BinData::Primitive
+    endian :big
+    search_prefix :keytab
+
+    # @!attribute [rw] len
+    #   @return [Integer]
+    uint16 :len, value: -> { data.length }
+
+    # @!attribute [rw] data
+    #   @return [String]
+    string :data, read_length: :len
+
+    def get
+      data.snapshot
+    end
+
+    def set(v)
+      self.data = v
+    end
+  end
+
+  class KeytabKeyblock < BinData::Record
+    endian :big
+    search_prefix :keytab
+
+    # @!attribute [rw] enctype
+    #   @return [Integer] The encryption type
+    # @see Rex::Proto::Kerberos::Crypto::Encryption
+    uint16 :enctype
+
+    # @return [KeytabCountedOctetString]
+    counted_octet_string :data
+  end
+
+  class KeytabEpoch < BinData::Primitive
+    endian :big
+    search_prefix :keytab
+
+    # @!attribute [rw] epoch
+    #   @return [Integer]
+    uint32 :epoch
+
+    def get
+      Time.at(epoch)
+    end
+
+    def set(v)
+      self.epoch = v.to_i
+    end
+  end
+
+  class KeytabEntry < BinData::Record
+    endian :big
+    search_prefix :keytab
+
+    # @return [Integer] The number of bytes for the len field
+    LEN_FIELD_BYTE_SIZE = 4
+
+    # @!attribute [rw] len
+    #   @return [Integer] The number of remaining bytes for this record. The length does not include the 4 bytes for this field
+    int32 :len,
+          value: -> {
+            size = [
+              count_of_components,
+              realm,
+              components,
+              name_type,
+              timestamp,
+              vno8,
+              keyblock,
+              # TODO: Omit vno and flags to ensure we generate MIT output for cross-compatibility with ktpass and wireshark
```

Comment:
when you say omit do you mean from the model? from the output? only omit in certain cases?
I'm a bit lost on the reasoning here

Reply:
Left as a TODO as a discussion point for module hacking 🕵️